heat/core/communication.py

```diff
@@ -463,6 +464,161 @@ def Iallgatherv(self, sendbuf, recvbuf, send_axis=0, recv_axis=None):
         return self.__allgather_like(self.handle.Iallgatherv, sendbuf, recvbuf, send_axis, recv_axis)
     Iallgatherv.__doc__ = MPI.Comm.Iallgatherv.__doc__
 
+    @classmethod
+    def alltoall_sendbuffer(cls, obj, nproc, counts=None, displs=None): #This is a temporary soltion to pass number of arguments as variable. Better to access via cls itself, but how?
```

Comment:
I do not understand the comment given.
What would you like to access via the class directly? counts and displs? If so, they are perfectly fine as parameters. If not, what is it?

Reply:
The problem is nproc (the number of processes). For the other functions, you can just do self.size. But with the classmethod this doesn't work. Is there some way to do cls.size() ? otherwise it has to be passed as a parameter. I believe counts and displs are not even needed

Reply:
You can change it to be a simple instance method, aka

def alltoall_sendbuffer(self, obj, ...):

Reply:
oh ok, wasn't sure (I still don't get a lot of python concepts (like classmethod )). This might also make sense since these functions are only used for all_to_all (unlike the very versatile as_buffer)

Reply:
argh, but can I then still access __mpi_type_mappings? 
   @classmethod
    def alltoall_sendbuffer(cls, obj, nproc, counts=None, displs=None): #This is a temporary soltion to pass number of arguments as variable. Better to access via cls itself, but how?
        mpi_type, elements = cls.__mpi_type_mappings[obj.dtype], torch.numel(obj)

Reply:
Yes, with self.__mpi_type_mappings